sigma/types.py

```diff
@@ -706,9 +705,26 @@ class SigmaRegularExpression(SigmaType):
         SigmaRegularExpressionFlag.DOTALL: "s",
     }
 
-    def __post_init__(self):
+    def __init__(
```

Comment:
Is there a specific reason to use `__init__` instead of `__post_init__`? A string `regexp` can also be replaced in `__post_init__`.